yourbench/utils/loading_engine.py

```diff
@@ -1,86 +1,53 @@
-        {'a': 'bar', 'b': ['bar', 123]}
-    """
+    """Recursively expand environment variables in nested structures."""
     if isinstance(obj, dict):
         return {k: _expand_env_vars(v) for k, v in obj.items()}
     if isinstance(obj, list):
-        return [_expand_env_vars(item) for item in obj]
+        return [_expand_env_vars(i) for i in obj]
     if isinstance(obj, str):
         return os.path.expandvars(obj)
     return obj
 
 
-def load_config(config_path: str) -> Dict[str, Any]:
-    """
-    Load the task configuration from a YAML file, substituting environment variables.
-
-    This function reads a YAML configuration file, expands any environment variables
-    present (using the '$VAR' syntax), and returns the configuration as a dictionary.
-    Environment variable substitution is performed recursively on all string values
-    in the resulting configuration dictionary.
-
-    Parameters:
-        config_path (str): Path to the YAML configuration file.
-
-    Returns:
-        Dict[str, Any]: The configuration loaded as a dictionary.
-
-    Raises:
-        FileNotFoundError: If the configuration file could not be found at config_path.
-        yaml.YAMLError: If there was an error parsing the YAML content.
-    """
-    # Load environment variables from .env files
-    load_dotenv()
-
-    if not os.path.exists(config_path):
-        logger.error("Configuration file not found: {}", config_path)
-        raise FileNotFoundError(f"Configuration file not found: {config_path}")
-
-    try:
-        # Read the raw configuration file
-        with open(config_path, "r") as file:
-            config_str = file.read()
-        logger.debug("Successfully read configuration file from {}", config_path)
-
-        # Substitute environment variables in the configuration string
-        expanded_config_str = os.path.expandvars(config_str)
-
-        # Parse the YAML configuration
-        config = yaml.safe_load(expanded_config_str)
-
-        # Recursively expand environment variables in all string values
-        config = _expand_env_vars(config)
-
-        logger.info("Configuration loaded successfully from {}", config_path)
-        return config
-
-    except Exception as exc:
-        logger.exception("Failed to load configuration due to: {}", exc)
-        raise
+def load_config(config_path: str | Path) -> dict[str, Any]:
+    """Convenience wrapper returning ``ConfigLoader(config_path).load()``."""
+    return ConfigLoader(Path(config_path)).load()
+
+
+@dataclass(slots=True)
+class ConfigLoader:
+    path: Path
+
+    def load(self) -> dict[str, Any]:
+        load_dotenv()
+
+        if not self.path.is_file():
+            logger.error(f"Configuration file not found: {self.path}")
+            raise FileNotFoundError(self.path)
+
+        try:
+            text = self.path.read_text()
+            logger.debug(f"Read configuration from {self.path}")
+            expanded = os.path.expandvars(text)
```

Comment:
add back the command to explain what the expandvars does as it's not that common